### PR TITLE
Rack and Sinatra - working

### PR DESCRIPTION
**Pull request by almuredin**
Working sinatra

### CODE REVIEW

File: config.ru [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require "bundler/setup"
+
+require_relative "./shop"
+
+
```
Comment (gotar) [inline config.ru]: those one empty line

File: lib/services/add_to_basket.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 module Shop
   class AddToBasket
+    def initialize(params)
+      @product_id = params.fetch("product_id")
+      @quantity = params.fetch("quantity")
+    end
     def call(product, quantity)
```
Comment (gotar) [inline lib/services/add_to_basket.rb]: add empty line between methods

File: lib/services/add_to_basket2.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Shop
+  class AddToBasket2
```
Comment (gotar) [inline lib/services/add_to_basket2.rb]: Hmm you should decide to one class, having very similar 2 will not help for sure

File: lib/services/fetch_product.rb [hunk 4]
```diff
@@ -1,8 +1,9 @@
 module Shop
   class FetchProduct
     def call(id)
```
Comment (gotar) [inline lib/services/fetch_product.rb]: product_id on place id will be more descriptive

File: views/basket/show.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  <tbody>
+    <% basket.each do |b| %>
+    <tr>
+      <td><%= b[:product].name %></td>
+      <td><%= b[:quantity] %></td>
+      <td><%= b[:product].netto %></td>
+      <td><%= b[:product].vat %></td>
+      <td><%= b[:sum_for_product] %></td>
+    </tr>
+    <% end %>
```
Comment (gotar) [inline views/basket/show.erb]: wrong indentation, everything inside do block should be indented

File: views/layout.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+</head>
+
+<body>
+<h1> My lovely shop </h1>
+<a href="/">Home</a> |
+<a href="/basket">Basket</a>
+<%= yield %>
+
+  </body>
+</html>
```
Comment (gotar) [inline views/layout.erb]: wrong indentation in file

File: views/products/index.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<h1>Products:</h1>
+  <% products.each do |product| %>
```
Comment (gotar) [inline views/products/index.erb]: wrong indentation

File: Gemfile [hunk 8]
```diff
@@ -1,5 +1,8 @@
 source "https://rubygems.org"
 
+#Web framework
```
Comment (smt116) [inline Gemfile]: this comment is unnecessary

Comment (gotar) [inline Gemfile]: I like to group gems in big project, for example:
https://github.com/gotar/dry-web-skeleton/blob/master/Gemfile

Comment (gotar) [inline Gemfile]: Good practice for me

Comment (smt116) [inline Gemfile]: 👍 for grouping, 👎 for those comments, but okay :P

Comment (gotar) [inline Gemfile]: yeap I comment gems when I work, and then group if needed and load in test, files only gems I need (from specific group) but I can agree that here (where we have only few gems) it's pointless

Comment (gotar) [inline Gemfile]: But it's something I show during hangout so it's my "mistake"

**Pull request merged**